### PR TITLE
More csrf tokens

### PR DESCRIPTION
**Pull request by calellowitz**
@TylerSheffels @sravfeyn cc: @gcapalbo 
I added to CSRF tokens to forms that have failed on staging due to CSRF protection, and added the ajax CSRF headers to a few more places, where those requests were failing. It can probably be reviewed all at once. 

On a related note, it has become pretty clear that the logging on staging is only catching a small subset of forms that will break if we implement CSRF protection on prod. Its missing some pretty important things too. I discovered out proper CSRF protection would have broken almost all of the form actions, but the staging logs only covered changing the settings because staging isn't used widely enough to catch all cases. Unless there are strong objections, I really think we should merge the CSRF logging code to master and look at prod logs in a week or so, because we are almost certainly missing a ton of broken forms.

### CODE REVIEW

File: corehq/apps/app_manager/templates/app_manager/form_designer.html [hunk 1]
```diff
@@ -96,7 +96,8 @@
                     return $('header').outerHeight() + 
                         $('.hq-page-header-container').outerHeight();
                 }
-            }
+            },
+            csrftoken: $.cookie('csrftoken'),
```
Comment (calellowitz) [inline corehq/apps/app_manager/templates/app_manager/form_designer.html]: The form builder save request originates in Vellum, so I am passing the CSRF token to vellum when we instantiate the form builder. Vellum handling of this was added in https://github.com/dimagi/Vellum/pull/578

**Comment by czue** [comment]
> Unless there are strong objections, I really think we should merge the CSRF logging code to master and look at prod logs in a week or so, because we are almost certainly missing a ton of broken forms.

sounds good to me.

File: corehq/apps/settings/templates/settings/bootstrap2/base_template.html [hunk 2]
```diff
@@ -2,6 +2,7 @@
 {% load hq_shared_tags %}
 
 {% block js %} {{ block.super }}
+    <script type="text/javascript" src="{% new_static 'hqwebapp/js/ajax_csrf_setup.js' %}"></script>
```
Comment (czue) [inline corehq/apps/settings/templates/settings/bootstrap2/base_template.html]: should this be `static`?

**Comment by czue** [comment]
looks good to me except the one question. does just including `ajax_csrf_setup.js` on the page automatically add it to all ajax requests? might be good if there were docs somewhere about how to set this up moving forwards.

**Comment by czue** [comment]
(the non-standard django parts, that is)

**Comment by calellowitz** [comment]
@czue It does for jquery ones. Including the script overwrites the jquery ajax function, to add a the CSRF token to the header of the request. It does not change ajax requests that originate from angularjs code. There it has to be done on a per app level. I can definitely add documentation for this work, is confluence the right place? or tagging team commcarehq with an explainer here?

**Comment by czue** [comment]
I like readme files or python-style docs in modules, but no strong opinions. there's also [read the docs](http://commcare-hq.readthedocs.org/en/latest/)

**Comment by calellowitz** [comment]
@czue i think you were right about the static vs new static question, but i also pushed the import of that file to the base template, so it should be on all pages bootstrap2 pages now and cover a lot more cases we dont yet know we are missing. We also shouldnt have to worry about importing it manually as much now.
read the docs looks like good place for documentation about this, i will start writing that up.

**Comment by sravfeyn** [comment]
No concerns from me to put it on prod now. Wanted to get frequent failures fixed first by putting it on staging first.

Also re. `ajax_csrf_setup.js` the only caveat is that if an ajax call itself is overriding `beforesend`, it should call the parent `beforesend` which adds csrf header.

**Pull request merged**